third_party/amd/lib/AMDGPUToLLVM/AMDGPUToLLVMPass.cpp

```diff
@@ -0,0 +1,79 @@
+#include "AMDGPUToLLVM/AMDGPUToLLVMPass.h"
+#include "Dialect/AMDGPU/IR/Dialect.h"
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Support/LogicalResult.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+#include "triton/Analysis/AxisInfo.h"
+#include "triton/Conversion/TritonGPUToLLVM/TypeConverter.h"
+// clang-format off
+#include "triton/Dialect/TritonGPU/Transforms/Utility.h"
+#include "amd/lib/TritonAMDGPUToLLVM/Utility.h"
+// clang-format on
+
+using namespace mlir;
+using namespace mlir::triton;
+using namespace mlir::triton::gpu;
+namespace tta = mlir::triton::amdgpu;
+
+#define GEN_PASS_CLASSES
+#include "AMDGPUToLLVM/Passes.h.inc"
+
+namespace mlir::triton::AMD {
+
+class AMDDialectLLVMConversionTarget : public ConversionTarget {
+public:
+  explicit AMDDialectLLVMConversionTarget(MLIRContext &ctx)
+      : ConversionTarget(ctx) {
+    addLegalDialect<LLVM::LLVMDialect>();
+    addLegalDialect<ROCDL::ROCDLDialect>();
+    addLegalDialect<mlir::scf::SCFDialect>();
+    addLegalDialect<triton::TritonDialect>();
+    addLegalDialect<triton::gpu::TritonGPUDialect>();
+    addIllegalDialect<triton::nvidia_gpu::TritonNvidiaGPUDialect>();
```

Comment:
For illegal ones, maybe use `markUnknownOpDynamicallyLegal` to return false for all of them? This way we are a bit future proof so that we don't need to remember updating this after dialect changes.